### PR TITLE
Feature 22. Leave feedback to restaurant as user. narek

### PR DESCRIPTION
**Pull request by narekacy**
Merged dev. Check 'send a feedback for restaurant' and previous 'aitenticate' feature. 

Also improved regex.

### CODE REVIEW

File: server/src/main/java/com/lunchtime/controllers/FeedbackController.java [hunk 1]
```diff
@@ -1,34 +1,45 @@
...
 import com.lunchtime.models.Feedback;
-import com.lunchtime.service.impl.FeedbackServiceImpl;
+import com.lunchtime.service.FeedbackService;
+import lombok.RequiredArgsConstructor;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
-//TODO remove redundant lines
+import javax.validation.Valid;
 @RestController
```
Comment (VozNick) [inline server/src/main/java/com/lunchtime/controllers/FeedbackController.java:15]: needs to be empty line)
PS. checkstyle error

**Review by Andrii35 — DISMISSED** [review]

File: server/src/main/java/com/lunchtime/controllers/FeedbackController.java [hunk 2]
```diff
@@ -1,34 +1,45 @@
...
 public class FeedbackController {
 
-    //TODO rename variable name to normal CamelCase
-    private final FeedbackServiceImpl feedbackServiceimplementation;
+    private final FeedbackService feedbackService;
 
-    public FeedbackController(FeedbackServiceImpl feedbackServiceimplementation) {
-        this.feedbackServiceimplementation = feedbackServiceimplementation;
+    @PostMapping
+    public ResponseEntity<FeedbackDto> saveFeedback(@Valid @RequestBody FeedbackDto feedbackDto) throws URISyntaxException {
```
Comment (VozNick) [inline server/src/main/java/com/lunchtime/controllers/FeedbackController.java]: line to long, checkstyle error

Comment (narekacy) [reply server/src/main/java/com/lunchtime/controllers/FeedbackController.java]: I had no warning. Fixed yet.

Comment (Andrii35) [reply server/src/main/java/com/lunchtime/controllers/FeedbackController.java]: You will laugh, but the checkstyle indicates that the parenthesis should be immediately after the method name. " /app/src/main/java/com/lunchtime/controllers/FeedbackController.java:24: 'method def lparen' has incorrect indentation level 8, expected level should be 4. [Indentation]" and "/app/src/main/java/com/lunchtime/controllers/FeedbackController.java:24:9: '(' should be on the previous line. [MethodParamPad]"

Comment (narekacy) [reply server/src/main/java/com/lunchtime/controllers/FeedbackController.java]: Fixed

**Review by VozNick — APPROVED** [review]

**Review by Andrii35 — APPROVED** [review]

**Review by vitaliimak — COMMENTED** [review]
image

There is some strange video artifact. Some flash
https://www.loom.com/share/94aacfcd58d34c4fae63a9c3187d44db

**Review by vitaliimak — CHANGES_REQUESTED** [review]
image
image

**Comment by narekacy** [comment]
@vitalii-mak, actually, I can't implement `shouldComponentMount()` yet, but If you know how it to do, I will learn it with pleasure

As for regex, I think it is better to keep some punctual policy here. If we let users to do what you are asking for, we increase risk that comments can be non-normal syntaxed

**Comment by VozNick** [comment]
Does this PR is going to be merge?

File: server/src/main/java/com/lunchtime/controllers/FeedbackController.java (hunk 1 above)
Comment (Melnyk9506) [reply server/src/main/java/com/lunchtime/controllers/FeedbackController.java:15]: checkstyle error still have please fix it

**Review by Melnyk9506 — APPROVED** [review]
i think this is a working option but we will need to create bug fixes because if you are not logged in then you basically should not see cancel button , you will also need to fix the moment with text input so that you can enter different characters in different languages and especially symbol for switching to another line

**Review by vitaliimak — COMMENTED** [review]
Hi

**Review by vitaliimak — APPROVED** [review]
(

**Pull request merged**